Refactor hipBLAS documentation (#950)
ROCm/hipBLAS: docs/install/Windows_Install_Guide.rst

.. meta::
  :description: Installing and Building hipBLAS for Windows
  :keywords: hipBLAS, rocBLAS, BLAS, ROCm, API, Linear Algebra, documentation, Windows installation, build

.. _windows-install:

***********************************
Installing and building for Windows
***********************************

> This topic covers how to install hipBLAS on Windows from a package and how to build and install it from the source code.

Reviewer: ```suggestion
This topic covers how to install hipBLAS on Microsoft Windows from a package and how to build and install it from the source code.
```

For a list of installation prerequisites, see :doc:`hipBLAS prerequisites <prerequisites>`.

> Installing pre-built packages

Reviewer: ```suggestion
Installing prebuilt packages
```

Reviewer: search page for this.

=============================

> hipBLAS can be installed on Windows 10 or 11 using the AMD HIP SDK installer.

Reviewer: link to HIP SDK?

The simplest way to use hipBLAS in your code is to use CMake. To install hipBLAS on Windows, follow these steps:

#. Add the SDK installation location to ``CMAKE_PREFIX_PATH`` in the CMake configuration command:

   ::

      -DCMAKE_PREFIX_PATH="C:\hipSDK"

#. Add the following lines to ``CMakeLists.txt``:

   ::

      find_package(hipblas)
      target_link_libraries( your_exe PRIVATE roc::hipblas )

Building and installing hipBLAS
===============================

Most users do not need to build hipBLAS from source because it can be used after installing the pre-built packages as described above.
If necessary, users can follow these instructions to build hipBLAS from source.

Download the hipBLAS source code
--------------------------------

The hipBLAS source code is available on the `hipBLAS GitHub <https://github.com/ROCm/hipBLAS>`_.

> The version of the ROCm HIP SDK might be shown in the default installation path,
> but you can run the HIP SDK compiler to display the version from the ``bin/`` folder:

Reviewer: ```suggestion
The ROCm HIP SDK version might be shown in the default installation path,
but you can run the HIP SDK compiler to display the version from the ``bin/`` folder:
```

This corresponds to major release = ``5``, minor release = ``4``, patch = ``22880``, and the build identifier ``135e1ab4``.
The hipBLAS GitHub contains branches with names like ``release/rocm-rel-major.minor``. Select the branch where the major and minor
fields correspond to the HIP version.
For example, use the following command format to download hipBLAS:

::

    git clone -b release/rocm-rel-x.y https://github.com/ROCm/hipBLAS.git

Replace ``x.y`` in this command with the version of the HIP SDK installed on your machine.
For example, if you have HIP 6.2 installed, then use ``-b release/rocm-rel-6.2``.
You can add the SDK tools to your path with an entry such as:

::

    %HIP_PATH%\bin

Building the library and library dependencies
---------------------------------------------

The root of the hipBLAS repository includes a helper Python script named ``rmake.py`` that lets you build and install hipBLAS
with a single command. It accepts several options but has a hard-coded configuration
that you can override by invoking ``cmake`` directly. However, it's a great way to get started quickly and serves
as an example of how to build and install hipBLAS.

> A few commands in the script require administrator access, so it might prompt you for a password.

Reviewer: ```suggestion
A few commands in the script require administrator access, so you might be prompted for a password.
```

Reviewer: not sure if any better
